libs/Database.js

```diff
@@ -462,14 +534,39 @@ class Database {
         const finalTableName = `${contract}_${table}`;
         const tableData = await this.getContractCollection(contract, finalTableName);
 
+        const contractInDb = await this.findContract({ name: contract });
```

Comment:
getContractCollection immediately above also does a findContract.  To avoid doing findContract twice, it would be more efficient to modify getContractCollection to return an object with both the tableData and primaryKey.